packages/purpose-process/test/testDeletePurpose.ts

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { describe, expect, it } from "vitest";
+import {
+  DraftPurposeDeletedV2,
+  Purpose,
+  PurposeId,
+  PurposeVersion,
+  WaitingForApprovalPurposeDeletedV2,
+  generateId,
+  purposeVersionState,
+  toPurposeV2,
+  toReadModelEService,
+} from "pagopa-interop-models";
+import {
+  getMockPurpose,
+  writeInReadmodel,
+  readLastEventByStreamId,
+  decodeProtobufPayload,
+  getMockPurposeVersion,
+} from "pagopa-interop-commons-test";
+import {
+  purposeNotFound,
+  organizationIsNotTheConsumer,
+  purposeCannotBeDeleted,
+} from "../src/model/domain/errors.js";
+import {
+  eservices,
+  postgresDB,
+  purposeService,
+  purposes,
+} from "./purposeService.integration.test.js";
+import { addOnePurpose, getMockEService } from "./utils.js";
+
+export const testDeletePurpose = (): ReturnType<typeof describe> =>
+  describe("deletePurpose", () => {
+    it("should write on event-store for the deletion of a purpose (no versions)", async () => {
+      const mockEService = getMockEService();
+      const mockPurpose: Purpose = {
+        ...getMockPurpose(),
+        eserviceId: mockEService.id,
+        versions: [],
+      };
+
+      await addOnePurpose(mockPurpose, postgresDB, purposes);
+      await writeInReadmodel(toReadModelEService(mockEService), eservices);
+
+      await purposeService.deletePurpose({
+        purposeId: mockPurpose.id,
+        organizationId: mockPurpose.consumerId,
+        correlationId: generateId(),
+      });
+
+      const writtenEvent = await readLastEventByStreamId(
+        mockPurpose.id,
+        "purpose",
+        postgresDB
+      );
+
+      expect(writtenEvent).toMatchObject({
+        stream_id: mockPurpose.id,
+        version: "1",
+        type: "DraftPurposeDeleted",
```

Comment:
```suggestion
        type:
            mockPurpose.versions.length === 0 ||
            mockPurpose.versions[0].state === purposeVersionState.draft
              ? "DraftPurposeDeleted"
              : "WaitingForApprovalPurposeDeleted",
```
Needed in order to make it the `it.each` work

Reply:
I prefer to keep them separated. The test should be as easy to read as possible.